### PR TITLE
fix(evpn-bridge): make tracer optional

### PR DESCRIPTION
**Pull request by mardim91**
Make tracer optional so we can import the netlinkWrapper library from evpn-gw-cni. 

More info here: https://github.com/opiproject/opi-gateway-evpn-cni/pull/14#discussion_r1515365216

### CODE REVIEW

File: pkg/utils/netlink.go [hunk 1]
```diff
@@ -35,6 +36,12 @@ type Netlink interface {
 	LinkSetDown(context.Context, netlink.Link) error
 	LinkSetMaster(context.Context, netlink.Link, netlink.Link) error
 	LinkSetNoMaster(context.Context, netlink.Link) error
+	LinkSetNsFd(context.Context, netlink.Link, int) error
```
Comment (artek-koltun) [inline pkg/utils/netlink.go:39]: The PR descriptions says that we are making the tracer optional. However, we are adding new methods. Should it be a part of a separate commit?

File: pkg/utils/netlink.go [hunk 2]
```diff
@@ -73,148 +80,261 @@ type NetlinkWrapper struct {
...
-	return &NetlinkWrapper{tracer: otel.Tracer("")}
+	return NewNetlinkWrapperWithArgs(true)
+}
+
+// NewNetlinkWrapperWithArgs creates initialized instance of NetlinkWrapper
+// based on passing arguments
+func NewNetlinkWrapperWithArgs(enableTracer bool) *NetlinkWrapper {
+	netlinkWrapper := &NetlinkWrapper{}
+	if enableTracer {
+		netlinkWrapper.tracer = otel.Tracer("")
```
Comment (artek-koltun) [inline pkg/utils/netlink.go:91]: I see that we need to check a tracer all the time here and there...
Is it possible to use [NoopTracerProvider](https://pkg.go.dev/go.opentelemetry.io/otel/api/trace#NoopTracerProvider) here, so we use that noop object and not checking for nil all the time?

File: pkg/utils/netlink.go (hunk 1 above)
Comment (mardim91) [reply pkg/utils/netlink.go:39]: Hello @artek-koltun,

I will prefer to have those changes if it is possible to a single PR as it is easier for me to test the CNI [PR](https://github.com/opiproject/opi-gateway-evpn-cni/pull/14) against this one. Mainly these changes that are introduced in this PR have as purpose to make the netlink and FRR wrapper libraries to be reused by other components like the [EVPN GW CNI](https://github.com/opiproject/opi-gateway-evpn-cni/pull/14)

I can update the description of the PR if that is ok but I would prefer to not have multiple PRs as it is difficult to test the CNI against this. Hope that is ok.

File: pkg/utils/netlink.go (hunk 2 above)
Comment (mardim91) [reply pkg/utils/netlink.go:91]: That is a very good point. Let me take a look

File: pkg/utils/netlink.go (hunk 1 above)
Comment (artek-koltun) [reply pkg/utils/netlink.go:39]: What if you test locally, then split and provide one PR and then another one?

Comment (mardim91) [reply pkg/utils/netlink.go:39]: ok let me do the changes on this PR and test them. If they are working I will discard this PR and create two individual ones.

Comment (mardim91) [reply pkg/utils/netlink.go:40]: Hello @artek-koltun,

I have made the changes with the noop tracer and tested them. They are working. I will create the two individuals PRs tommorow as we have discussed above

Comment (mardim91) [reply pkg/utils/netlink.go]: Hello @artek-koltun,

I have splitted the changes in two PRs. The first one is this PR here and the second one is [this](https://github.com/opiproject/opi-evpn-bridge/pull/383)

Please review when you find some time.

Comment (artek-koltun) [reply pkg/utils/netlink.go]: Could I ask you to update the storage bridges as well?

**Review by artek-koltun — APPROVED** [review]
LGTM

**Comment by mardim91** [comment]
Thank you @artek-koltun
